### PR TITLE
Make FeatureExtractor injectable

### PR DESCRIPTION
**Pull request by Grogdunn**
I want to make injectable FeatureExtractor because I don't need DJL. So everyone now can implement FeatureExtractor or use the provided DefaultFeatureExtractor.

Also removed `@Component` to avoid auto discovery (the bean is declared in configuration).

### CODE REVIEW

**Comment by bsbodden** [comment]
Thank you @Grogdunn I agree this is a good idea; I tried to make it as unintrusive as possible but this is even better. I'll review soon.

File: demos/roms-vss/src/main/resources/application.properties
Comment (bsbodden) [inline demos/roms-vss/src/main/resources/application.properties:1]: I think you left some settings behind...

**Review by bsbodden — CHANGES_REQUESTED** [review]
Look good, if you can just remove the extra settings in the demo. Thanks!

**Comment by Grogdunn** [comment]
removed my configuration ^^ sorry for that!

**Review by bsbodden — APPROVED** [review]
LGTM

**Pull request merged**